File: backend/banking-service/src/main/java/com/finix/notification/dto/NotificationEvent.java
package com.finix.notification.dto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class NotificationEvent {

    private Long customerId;

    private String eventType;

    private String title;

    private String message;

    private String email;

    private List<String> channels;

}

File: backend/notification-service/EbankingNotificationService/DTO/NotificationEvent.cs
namespace EbankingNotificationService.DTOs
{
    public class NotificationEvent
    {
        public int CustomerId { get; set; }

        public string EventType { get; set; } = string.Empty;

        public string Title { get; set; } = string.Empty;

        public string Message { get; set; } = string.Empty;

        // Optional because every notification does not need email/SMS
        public string? Email { get; set; }

        // Example: ["IN_APP", "EMAIL"]
        public List<string> Channels { get; set; } = new();
    }
}